app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -173,12 +201,31 @@ export async function search({
 }
 
 /** Get a circular by ID. */
-export async function get(circularId: number): Promise<Circular> {
+export async function get(
+  circularId: number,
+  version?: number
+): Promise<Circular> {
   if (isNaN(circularId)) throw new Response(null, { status: 404 })
   const db = await tables()
-  const result = await db.circulars.get({
-    circularId,
-  })
+
+  let result = version
+    ? await db.circulars_history.get({
+        circularId,
+        version,
+      })
+    : await db.circulars.get({
+        circularId,
+      })
+
+  if (!result && version !== undefined && !version && !isNaN(version)) {
+    result = await db.circulars.get({
+      circularId,
+    })
+  }
+
+  if (version && result?.version !== version) {
+    result = undefined
+  }
```

Comment:
So, the first ternary expression here still isn't quite right if we want a value of `undefined` to trigger lookup of the latest version but `0` or `NaN` to trigger an error.

Reply:
I see, do we just want the checks to short out in those cases? So move the check for If `0` or `NaN` ... up before doing any gets?

Reply:
You could... or you could structure the body of the function like this.

```ts
if (version === undefined) {
  // get most recent version
} (else if !version) {
  // throw 404
} else {
  // look up by version number
}
```

Reply:
Ah okay I didn't see this comment until after I pushed, what do you think of the latest refactor? It moves the errors up to handle it all at the beginning

Reply:
Ugh. You know what, if `version === 0` or `version === NaN`, I don't think you need to do anything special, since it will never match a real circular in the table.

Maybe this would be simpler using [function overloading](https://dev.to/zirkelc/how-to-overload-functions-in-typescript-22c3)?

```ts
export async function get(circularId: number) {
  ...
}

export async function get(circularId: number, version: number) {
  ...
}
```